tests/unit/archive_base_test.py

```diff
@@ -196,43 +197,34 @@ async def test_download_zipfile(mocker, bad_zipfile, good_zipfile):
 
 
 @pytest.mark.asyncio
-async def test_download_file(mocker, file_data):
+async def test_download_file(mocker, tmp_path):
```

Comment:
So this now downloads a real file with `requests`, and then tries to download it again with `AbstractDatasetArchiver.download_file`, and compares that they're the same - is that right?

And if I understand this all right, the point of ripping out the mocks is that we test that `ClientSession.get()` does what we think it does, vs. just assuming that the mocks are reasonable. Is that right also?

Seems smart to me - we gotta test the IO stuff somewhere, might as well test it in a very narrowly focused IO test. But we should change the docstring :)